codalab/worker/image_manager.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import threading
+import time
+import traceback
+from collections import namedtuple
+
+from codalab.worker.worker_thread import ThreadDict
+
+logger = logging.getLogger(__name__)
+# Stores the download state of a Docker image (also includes the digest being pulled, digest string,
+# DependencyStage and relevant status message from the download)
+ImageAvailabilityState = namedtuple('ImageAvailabilityState', ['digest', 'stage', 'message'])
+
+
+class ImageManager:
+    """
+    ImageManager is the base interface for all image managers (docker, singularity, etc.)
+    No actual logic is encapsulated in this class.
```

Comment:
Instead of this, just say it's an abstract class. Explain what an "image manager" is. Maybe also say which methods subclasses need to implement.